autocodesign/devportalclient/spaceship/profiles.go

```diff
@@ -174,34 +174,43 @@ func (c *ProfileClient) DeleteProfile(id string) error {
 
 // CreateProfile ...
 func (c *ProfileClient) CreateProfile(name string, profileType appstoreconnect.ProfileType, bundleID appstoreconnect.BundleID, certificateIDs []string, deviceIDs []string) (autocodesign.Profile, error) {
-	cmd, err := c.client.createRequestCommand("create_profile",
-		bundleIDIdentifierArgKey, bundleID.Attributes.Identifier,
-		certificateIDArgKey, certificateIDs[0],
-		profileNameArgKey, name,
-		profileTypeArgKey, string(profileType),
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	output, err := runSpaceshipCommand(cmd)
-	if err != nil {
-		return nil, err
+	for i := 0; i < 6; i++ {
```

Comment:
I probably do not know all of the details of this workaround and that is why I will be asking asking these questions. 

Why is this 6? It seems like an arbitrary high number to me. I mean in my mind I was comparing this to a scenario where the server returns a 500 error and we will still make 5 request just to be sure it is down. I know it is not a good example but could not come up with a better one. So retrying only once is not enough? Or even better did you find out why spaceship can end up returning an empty profile?

Another thing is that could this for loop be converted to some retry functionality we have in the util repos? I am not fully aware what feature set the current retry mechanism has so just asking.

Reply:
Did not find out the root cause :/. One retry is definitely not enough, seen at least 3 retry needed (https://app.bitrise.io/build/2e08da10-5813-475c-a50c-cf3164be6409#?tab=log). 5 could work, was just not confident that it is enough. The retry util suggestion makes sense, as it would wait for a few seconds (which may help).

Reply:
Knowing these I am happy to have a high number there. I might even say to round it up to 10 to make sure we catch all these cases. 

I prefer to have comments in code for such code pieces (the ones which is not clear at first look). So I would even suggest to add a comment above the retry which is something like:
```
// This retry is needed because spaceship can return an empty profile. In local testing at least 3 retries were needed in some cases. The {{insert final high retry number here}} retries probably will be never needed but only act as a big buffer to mitigate all occurrences.
```
I am not keen on the wording so feel free to rephrase it. 😄